xbmc/platform/posix/filesystem/SMBDirectory.cpp

```diff
@@ -116,7 +116,7 @@ bool CSMBDirectory::GetDirectory(const CURL& url, CFileItemList &items)
     int64_t size = 0;
     const bool isDir = S_ISDIR(st.st_mode);
     int64_t timeDate = 0;
-    bool hidden = StringUtils::StartsWith(name, ".");
+    bool hidden = name.front() == '.';
```

Comment:
I would prefer `name.starts_with('.')` as it is save in case `name` is empty for some weird reason.